https/https_proxy.go

```diff
@@ -0,0 +1,73 @@
+package https
+
+import (
+	"crypto/tls"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"time"
+)
+
+func handleTunneling(w http.ResponseWriter, r *http.Request) {
+	destConn, err := net.DialTimeout("tcp", r.Host, 10*time.Second)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	hijacker, ok := w.(http.Hijacker)
+	if !ok {
+		http.Error(w, "Hijacking not supported", http.StatusInternalServerError)
+		return
+	}
+	clientConn, _, err := hijacker.Hijack()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+	}
+	go transfer(destConn, clientConn)
+	go transfer(clientConn, destConn)
+}
+
+func transfer(destination io.WriteCloser, source io.ReadCloser) {
+	defer destination.Close()
+	defer source.Close()
+	io.Copy(destination, source)
+}
+
+func handleHTTP(w http.ResponseWriter, req *http.Request) {
+	resp, err := http.DefaultTransport.RoundTrip(req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		return
+	}
+	defer resp.Body.Close()
+	copyHeader(w.Header(), resp.Header)
+	w.WriteHeader(resp.StatusCode)
+	io.Copy(w, resp.Body)
+}
+
+func copyHeader(dst, src http.Header) {
+	for k, vv := range src {
+		for _, v := range vv {
+			dst.Add(k, v)
+		}
+	}
+}
+
+// StartHTTPSProxy starts a proxy that accepts to the HTTP CONNECT method to proxy arbitrary HTTPS requests
+func StartHTTPSProxy() {
+	server := &http.Server{
+		Addr: ":8888",
```

Comment:
This should be CLI configurable similar to [proxy-listen](https://github.com/PelionIoT/edge-proxy/blob/1e03559a81fe974e346598b5025862337c2c9081/cmd/edge-proxy/main.go#L55).

Reply:
Done